tests/unit/lms/views/dashboard/api/user_test.py

```diff
@@ -94,12 +91,94 @@ def test_students_metrics(  # pylint:disable=too-many-locals
                 g.authority_provided_id for g in segments
             ]
 
-        if with_auto_grading:
-            assignment.auto_grading_config = AutoGradingConfig(
-                activity_calculation="separate",
-                grading_type="all_or_nothing",
-                required_annotations=1,
+        db_session.flush()
+        user_service.get_users.return_value = select(User).where(
+            User.id.in_(
+                [u.id for u in [student, student_no_annos, student_no_annos_no_name]]
             )
+        )
+        dashboard_service.get_request_assignment.return_value = assignment
+        h_api.get_annotation_counts.return_value = annotation_counts_response
+
+        response = views.students_metrics()
+
+        dashboard_service.get_request_assignment.assert_called_once_with(
+            pyramid_request
+        )
+        h_api.get_annotation_counts.assert_called_once_with(
+            [g.authority_provided_id for g in assignment.groupings],
+            group_by="user",
+            resource_link_ids=[assignment.resource_link_id],
+            h_userids=sentinel.h_userids,
+        )
+        expected = {
+            "students": [
+                {
+                    "h_userid": student.h_userid,
+                    "lms_id": student.user_id,
+                    "display_name": student.display_name,
+                    "annotation_metrics": {
+                        "annotations": 4,
+                        "replies": sentinel.replies,
+                        "last_activity": sentinel.last_activity,
+                    },
+                },
+                {
+                    "h_userid": student_no_annos.h_userid,
+                    "lms_id": student_no_annos.user_id,
+                    "display_name": student_no_annos.display_name,
+                    "annotation_metrics": {
+                        "annotations": 0,
+                        "replies": 0,
+                        "last_activity": None,
+                    },
+                },
+                {
+                    "h_userid": student_no_annos_no_name.h_userid,
+                    "lms_id": student_no_annos_no_name.user_id,
+                    "display_name": None,
+                    "annotation_metrics": {
+                        "annotations": 0,
+                        "replies": 0,
+                        "last_activity": None,
+                    },
+                },
+            ]
+        }
+        assert response == expected
+
+    @pytest.mark.parametrize("with_last_grade", [True, False])
+    def test_students_metrics_with_auto_grading(  # pylint:disable=too-many-locals
```

Comment:
Same here in the test, new test case just for the autograding asserts, the  other test was getting to many conditions.

Agreeing with https://x.com/garybernhardt/status/1773443705716019338 but this was crossing the line.